### ISSUE TITLE
Add support for CockroachDB as a database option

### ISSUE THREAD

**Issue by eadoking** [umami-software/umami#2851]
### Describe the feature or enhancement

I propose adding support for [CockroachDB](https://www.cockroachlabs.com/) as a database option for Umami. CockroachDB is a cloud-native, distributed SQL database that offers several advantages for Umami users:

CockroachDB is wire-compatible with PostgreSQL, which Umami already supports.

CockroachDB offers up to 10 GiB of storage and 50M Request Units (RUs) per organization per month in its free tier.

Implementation Overview:

- Modify database migrations to be compatible with CockroachDB's syntax and features.
- Update database connection logic to support CockroachDB's connection string format ( if needed)
- Adjust any SQL queries that may require CockroachDB-specific optimizations (for my case, I didn't need to do it)
- Update documentation to include CockroachDB setup and configuration instructions.

For a hoby project or even a any-size website, it is very nice to have the database on CockroachDB instead of running a docker image, we have every thing separated and ready to use, and we reduce the load on the server.

I have already used it for a personal project, and works perfectly, this needs changes to the migrations and some small changes along the way.

I am willing to create a PR for the changes once the feature is approved.

**Comment by franciscao633** [comment]
I think we are shying away from supporting more dbs. There is a lot of overhead in supporting migrations, testing environments, minor differences in syntax/functions, etc.  We did have a PR to support CockroachDB in the past, but it was closed out.

https://github.com/umami-software/umami/pull/1579#issuecomment-1319116329
